src/lib.rs

```diff
@@ -36,6 +36,8 @@ pub mod api;
 pub mod app;
 pub mod async_read;
 pub mod buffers;
+#[cfg(feature = "codecs")]
+pub mod codec;
```

Comment:
Small nit, but almost every other module at this level uses a plural name, so `codecs` might be a better match.

Reply:
This was my first attempt as well.

Unfortunately `cargo` doesn't allow naming a feature with the same name of a dependency, and we already have `lib/codec` taking that name. We can rename it after `lib/codec` has been removed in #8918.

Reply:
Adressed in dafd8603a27aaf4ea11011f22ca02210a63646fd of #8918